_source/_posts/2019-10-25-intro-android-gradle.md

```diff
@@ -0,0 +1,392 @@
+* Merge your code and resources with those of your libraries
+* Cache resources for compilation performance
+* Connect it all to the Android SDK
+* Finally packaging it into the right format with the appropriate signature
+
+In true Android fashion, you can configure most aspects of Gradle's behavior by providing Gradle with different configuration options. Going further, lots of freely available plugins add features to Gradle: extended error reporting, custom app icons, integration with other systems, etc.  You can also use Groovy or Kotlin to create your own completely custom plugins.
+
+This article will introduce you to the essential aspects of what Gradle does and how you can configure it I'll also introduce you to a couple more advanced Gradle concepts.
+
+Your App will also use Okta to safely and easily authenticate users, so you can focus on your business while Okta gives you the security you need, plus services like:
+
+* User [Authentication](https://developer.okta.com/product/authentication/) and [Authorization](https://developer.okta.com/product/authorization/)
+* User-data storage
+* [Social login](https://developer.okta.com/authentication-guide/social-login/)
+* [Multi-factor authentication](https://developer.okta.com/use_cases/mfa/)
+
+This article follows an example Android app that authenticates a user with [Okta's SDK for Android](https://developer.okta.com/code/android/) and demonstrates Gradle in Android. Please clone and set it up yourself, so you can follow along.  
+What you will also need:
+
+* [A Free Okta Developer Account](https://developer.okta.com/signup/)
+* [Android Studio](https://developer.android.com/studio)
+
+Let's get started!
+
+## Set Up Your Android App to work With Okta and Gradle
+
+First, clone the [Okta Android sample app repo](https://github.com/okta/okta-oidc-android/blob/master/app/README.md) and open it with Android Studio:
+
+```bash
+git clone https://github.com/okta/okta-oidc-android
+cd okta-oidc-android
+```
+
+Next:
+
+1. Sign up for a free [Okta developer account](https://developer.okta.com/signup/)
+2. Create a new **Application** for **Native Android**
+3. Create a *config.json* file similar to this one in the *app* module's `/app/res/raw` folder:
+
+    ```json
+    {
+    "client_id": "{clientId}",
+    "redirect_uri": "{redirectUri}",
+    "end_session_redirect_uri": "{endSessionUri}",
+    "scopes": [
+        "openid",
+        "profile",
+        "offline_access"
+    ],
+    "discovery_uri": "https://{yourOktaDomain}"
+    }
+    ```
+
+4. Set a *URI Scheme* unique to your app in the *app* module's `/app/build.gradle`:
+
+    ```groovy
+    android.defaultConfig.manifestPlaceholders = [
+        "appAuthRedirectScheme": "com.okta.example"
+    ]
+    ```
+
+5. Add your Org URL (found in the upper right of the console home page) to your /local.properties file at the root of the project:
+
+```properties
+authn.orgUrl="https://dev-######.okta.com"
+```
```

Comment:
These properties are not well described `discovery_uri` uses `https://{yourOktaDomain}` and `orgUrl` uses `https://dev-######.okta.com`  (though the latter contains where to find that info and it _should_ be obvious) the above `config.json` block doesn't mention where the to grab the config values or what they should be.